src/app/embed/adapters/qsd.adapter.ts

```diff
@@ -0,0 +1,58 @@
+import { EMBED_AUDIO_URL_PARAM, EMBED_TITLE_PARAM, EMBED_SUBTITLE_PARAM,
+  EMBED_SUBSCRIBE_URL_PARAM, EMBED_SUBSCRIBE_TARGET, EMBED_IMAGE_URL_PARAM } from './../embed.constants';
+import { Observable } from 'rxjs/Observable';
+
+export class QSDAdapter {
+  constructor(private params: Object) {
+    this.params = params
+  }
+
+  get hasFullInformation(): boolean {
+    return (
+      (this.audioUrl != undefined) &&
+        (this.title != undefined) &&
+        (this.subtitle != undefined) &&
+        (this.subscribeUrl != undefined) &&
+        (this.subscribeTarget != undefined) &&
+        (this.artworkUrl != undefined)
+    );
+  }
+
+  get getParams(): Observable<Object> {
```

Comment:
Perfect. Again, probably a good idea to define an interface instead of using `Object` for the Observable type.